controlplane/kubeadm/internal/cluster.go

```diff
@@ -112,26 +112,11 @@ func (m *Management) GetWorkloadCluster(ctx context.Context, clusterKey client.O
 	restConfig = rest.CopyConfig(restConfig)
 	restConfig.Timeout = 30 * time.Second
 
-	if m.Tracker == nil {
```

Comment:
I think this nil check is pretty useless. If it would be a problem it would panic in l.108 at the moment. And if it would be problem our e2e tests would never be green...